code/game/mecha/mecha.dm

```diff
@@ -1072,6 +1072,11 @@
 	occupant_message("Toggled lights [lights ? "on" : "off"].")
 	log_message("Toggled lights [lights ? "on" : "off"].")
 
+/obj/mecha/extinguish_light(force)
+	if(!lights)
+		return
+	toggle_lights()
```

Comment:
Maybe worth adding a `msg=TRUE` default arg to `toggle_lights()`?

These 2 messages will run if the lights are extinguished, which makes it feel user invoked imho 
```dm
	occupant_message("Toggled lights [lights ? "on" : "off"].")
	log_message("Toggled lights [lights ? "on" : "off"].")
```

Reply:
done